File: codingjs/countClumps.js
function countClumps(nums){
  let count = 0;
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] === nums[i + 1]) {
      count++;
      for (let j = i + 2; j < nums.length; j++) {
        if (nums[i] !== nums[j]) break;
      }
      i = j - 1;
    }
  }
  return count;
}

File: codingbat/java/countClumps.java
public int countClumps(int[] nums) {
  if (nums.length < 2) return 0;
  int count = 0;
  int prevNum = nums[0];
  int clumpsCount = 0;
  for (int num: nums) {
    if (num == prevNum) {
      count++;
    } else {
      if (count >= 2) clumpsCount++;
      count = 1;
    }
    prevNum = num;
  }
  if (count >= 2) clumpsCount++;
  return clumpsCount;
}

